package/kernel/linux/modules/netdevices.mk

```diff
@@ -1457,3 +1457,18 @@ define KernelPackage/mhi-wwan-mbim/description
 endef
 
 $(eval $(call KernelPackage,mhi-wwan-mbim))
+
+define KernelPackage/aqc
+  SUBMENU:=$(NETWORK_DEVICES_MENU)
+  TITLE:=Aquantia AQtion (AQC107) 10Gbps Ethernet NIC
+  DEPENDS:=@PCI_SUPPORT +kmod-ptp
+  KCONFIG:=CONFIG_AQTION
+  FILES:=$(LINUX_DIR)/drivers/net/ethernet/aquantia/atlantic/atlantic.ko
+  AUTOLOAD:=$(call AutoProbe,aqc)
```

Comment:
I doubt this works as the resulting kmod is `atlantic.ko` and I dont see a `MODULE_ALIAS` in the driver

Reply:
@robimarko I dont get this - what should I be doing here or what the issue? (I want to understand you comment).

Reply:
My point is that `AutoProbe` is probably not working at all here, thus driver does not get autoloaded when needed.
It will match either the kmod name or if there is an alias which I doubt.

Has this been runtime tested at all?

Reply:
@robimarko - I'll retest after recent changes.

To be clear - I should just select drivers as module not `*` in kernel_menuconfig and then proceed to boot via image and observe if it automatically loaded right?

Reply:
No, no.
You should not be using `kernel_menuconfig` at all here, but rather select the kmod in `menuconfig`, because you were not even testing the packaged kmod, no wonder it worked with completely wrong value

Reply:
Hey @robimarko - I tested with both `atlantic` nameing as well as `aquantia` as per your ealier suggestions and either way the releavant entries seem to be there in - eg:

` /sys/module/atlantic/drivers/pci:atlantic/...`

And the actual interface works fine - but I'm not sure how else to verify - can you please advise.

Reply:
And you made sure that it was not built-in but rather a module?
Please post output of `lsmod`

Reply:
Hey @robimarko I've selected this as module in both `menuconfig` & `kernel_menuconfig` (as per images below) and I think the issue is with `AutoProbe` as you were saying earlier? - because I do not see the device loading or working in this case. Is this an issue or something I'm doing wrong still? - thanks again.
![image](https://user-images.githubusercontent.com/974854/187089890-6b3234e3-8773-4da6-b567-d31754c9c2b3.png)
![image](https://user-images.githubusercontent.com/974854/187089870-c338f0a7-ac59-4a71-ac6b-fb333fc8d6e2.png)
![image](https://user-images.githubusercontent.com/974854/187089930-72abe934-1924-404d-a758-add28a1c46b0.png)

Reply:
No, no.
Again, you should not be selecting it in `kernel_menuconfig` at all, just the kmod via menuconfig.
And yeah, if its not being loaded then `AutoProbe` is incorrect

Reply:
I restarted with a new clone such that:

```
git clone https://git.openwrt.org/openwrt/openwrt.git && cd openwrt ;
# // put 'atlantic' defintion in netdevices.mk then
./scripts/feeds update -a ; && ./scripts/feeds install -a ;
make menuconfig ;
# // Selected '*' for screenshot below. 
make -j $(nproc) clean defconfig download world V=sc ;
```

![image](https://user-images.githubusercontent.com/974854/187272603-72ffef25-a8c4-48d5-9a72-b829f0d9f325.png)

@robimarko - the lsmod outputs are as follows:

```shell
lsmod && uname -ar ; 
  # # // ...
  # atlantic              147456  0 
  # # // ...
  # Linux version 5.10.138 (vagrant@make-host) (x86_64-openwrt-linux-musl-gcc (OpenWrt GCC 11.3.0 r20430-18a2b29aa1) 11.3.0, GNU ld (GNU Binutils) 2.37) #0 SMP Mon Aug 29 09:54:00 2022
```